src/app/auth/signup.tsx

```diff
@@ -29,39 +109,87 @@ function SignUpScreen() {
 
   return (
     <View style={globalStyles.auth_container}>
+      <Text style={[globalStyles.h3, globalStyles.mt20]}>
+        Read stories from young creators
+      </Text>
+
+      <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
+        <TextInput
+          onChangeText={text => setUsername(text)}
+          value={username}
+          style={styles.inputField}
+          placeholder="Enter New Username"
+          autoCapitalize="none"
+        />
+        {usernameError && <Text style={styles.error}>{usernameError}</Text>}
+      </View>
+      <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
+        <TextInput
+          onChangeText={text => setFirstName(text)}
+          value={firstName}
+          style={styles.inputField}
+          placeholder="First Name"
+          autoCapitalize="none"
+        />
+      </View>
+      <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
+        <TextInput
+          onChangeText={text => setLastName(text)}
+          value={lastName}
+          style={styles.inputField}
+          placeholder="Last Name"
+          autoCapitalize="none"
+        />
+      </View>
+
       <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
-        <Input
-          label="Email"
-          leftIcon={{ type: 'font-awesome', name: 'envelope' }}
+        <TextInput
           onChangeText={text => setEmail(text)}
           value={email}
-          placeholder="email@address.com"
+          style={styles.inputField}
+          placeholder="Enter Email"
           autoCapitalize="none"
+          textContentType="emailAddress"
         />
       </View>
+      {emailError && <Text style={styles.error}>{emailError}</Text>}
 
-      <View style={globalStyles.verticallySpaced}>
-        <Input
-          label="Password"
-          leftIcon={{ type: 'font-awesome', name: 'lock' }}
+      <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
+        <TextInput
           onChangeText={text => setPassword(text)}
           value={password}
-          secureTextEntry
-          placeholder="Password"
+          style={styles.inputField}
+          placeholder="Enter Password"
           autoCapitalize="none"
+          secureTextEntry={true}
+          textContentType="password"
         />
+      </View>
 
-        <Link href="/auth/login">Already have an account? Log In</Link>
-        <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
-          <Button
-            title="Sign Up"
-            disabled={loading}
-            onPress={signUpWithEmail}
-          />
-        </View>
+      <Link href="/auth/login">Already have an account? Log In</Link>
+      <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
+        <Button
+          title="Sign Up"
+          disabled={loading || emailError != '' || usernameError != ''}
+          onPress={signUpWithEmail}
+        />
       </View>
```

Comment:
Let's make a button component that's styled according to the designs so we can use it across the onboarding & login flow